### PR TITLE
Bug: fix generate title function

### PR DESCRIPTION
**Pull request by ianlcassidy**
### Motivation and Context

The `generate_title` function in `app.py` has a few bugs in it.

### Description

- remove `use_data` kwarg since `init_openai_client()` no long accepts it
- remove prompt instructions around outputting the title in JSON format
  - this seems unnecessarily complex and can lead to JSON decode errors
  - just output the title as plain text
- add exception logging in the function to capture when something fails
- also added `DATASOURCE_TYPE` to `.env.sample` since that controls what data source the tool should use
  - should this default to `AzureCognitiveSearch` either in `.env.sample` or  `backend.settings`?

### Contribution Checklist

- [x] I have built and tested the code locally and in a deployed app
- [x] For frontend changes, I have pulled the latest code from main, built the frontend, and committed all static files.
- [x] This is a change for all users of this app. No code or asset is specific to my use case or my organization.
- [x] I didn't break any existing functionality :smile:

### CODE REVIEW

**Comment by ianlcassidy** [comment]
> @microsoft-github-policy-service agree

**Comment by ianlcassidy** [comment]
@microsoft-github-policy-service agree

File: app.py [hunk 1]
```diff
@@ -846,14 +846,15 @@ async def generate_title(conversation_messages):
     messages.append({"role": "user", "content": title_prompt})
 
     try:
-        azure_openai_client = init_openai_client(use_data=False)
```
Comment (sarah-widder) [inline app.py:849]: Is there a reason to remove this flag? It will work best without data.

Comment (ianlcassidy) [reply app.py:849]: the function definition no longer accepts this kwarg. i didn't check the history about when it was removed, but calling `init_openai_client` is throwing an error in the code as it's currently written. either the `use_data` flag should be added back to the function or it needs to be removed as i've suggested here.
https://github.com/microsoft/sample-app-aoai-chatGPT/blob/main/app.py#L98

Comment (sarah-widder) [reply app.py:850]: Ah - you're right. Good catch!

**Review by sarah-widder — APPROVED** [review]
Looks great. Thank you for the contribution!

**Comment by ianlcassidy** [comment]
@sarah-widder feel free to merge when you want, i don't have the ability to.

**Pull request merged**